core/templates/domain/question/QuestionObjectFactory.ts

```diff
@@ -23,9 +23,11 @@ import { State, StateBackendDict, StateObjectFactory }
   from 'domain/state/StateObjectFactory';
 import INTERACTION_SPECS from 'interactions/interaction_specs.json';
 import constants from 'assets/constants';
+import { Misconception } from 'domain/skill/MisconceptionObjectFactory';
+
 
 export interface QuestionBackendDict {
-  'id': string;
+  'id': string | null;
```

Comment:
Okay, but why is it set to `null` there? You need to get the conceptual understanding so that you can then write a comment here, explaining when exactly will the field be `null`.

Reply:
yes field never be used as null. use empty string instead, is it fine. if not help me to proceed further

Reply:
No, empty string is not a good replacement, the best thing to do is to refactor the code so that we don't need to use `null` at all. The second best thing is to use `null` with proper explanation.

Reply:
ditto (for, why usage of null)

Add TODO for remove the createDefaultQuestion so that full question can be created from start. No need of null then.